mlos_bench/mlos_bench/config/services/remote/azure/scripts/setup-rg/setup-rg.ps1

```diff
@@ -0,0 +1,115 @@
+
+param(
+    # Main control plane ARM template params
+    [Parameter(Mandatory=$True)]
+    [string] $controlPlaneArmParameters,
+    # Results DB ARM template params
+    [Parameter(Mandatory=$False)]
+    [string] $resultsDbArmParameters,
+    # Other params
+    [Parameter(Mandatory=$True)]
+    [string] $servicePrincipalName,
+    [Parameter(Mandatory=$True)]
+    [string] $resourceGroupName,
+    [Parameter(Mandatory=$True)]
+    [string] $certName,
+    [Parameter(Mandatory=$False)]
+    [int] $certExpirationYears = 1
+)
+
+# Provision resources into the resource group with ARM template
+Write-Output "Provisioning control plane resources..."
+$deploymentResults = az deployment group create `
+    --resource-group $resourceGroupName `
+    --template-file .\rg-template.json `
+    --parameters $controlPlaneArmParameters `
+    | ConvertFrom-JSON
+
+if (!$?) {
+    Write-Error "Error in provisioning control plane resources!"
+    return
+}
+
+# Conditional provisioning of results DB
+if ($resultsDbArmParameters) {
+    Write-Output "Provisioning results DB..."
+    $dbDeploymentResults = az deployment group create `
+        --resource-group $resourceGroupName `
+        --template-file ./results-db/mysql-template.json `
+        --parameters $resultsDbArmParameters `
+        | ConvertFrom-JSON
+
+    if (!$?) {
+        Write-Error "Error in provisioning results DB!"
+    }
+    else {
+        $dbName = $dbDeploymentResults.properties.outputs.dbName.value
+        $vmName = $deploymentResults.properties.outputs.vmName.value
+        $vmIpAddress = $deploymentResults.properties.outputs.vmIpAddress.value
+
+        # VM IP access for results DB
+        az mysql flexible-server firewall-rule update `
+            --resource-group $resourceGroupName `
+            --name $dbName `
+            --rule-name "AllowVM-$vmName" `
+            --start-ip-address $vmIpAddress `
+            --end-ip-address $vmIpAddress
+    }
+}
+
+$currentUserAlias = az account show --query "user.name" --output tsv
+$resourceGroupId = az group show --name $resourceGroupName --query "id" --output tsv
+
+# Assign 'Key Vault Administrator' access to current user
+$kvName = $deploymentResults.properties.outputs.kvName.value
+$kvId = az keyvault show --name $kvName --resource-group $resourceGroupName --query "id" --output tsv
+az role assignment create `
+    --assignee $currentUserAlias `
+    --role "Key Vault Administrator" `
+    --scope $kvId
+
+# Check if cert of same name exists in keyvault already
+$certThumbprint = az keyvault certificate show `
+    --name $certName `
+    --vault-name $kvName `
+    --query "x509ThumbprintHex" --output tsv `
+    2>$null
+
+if (!$?) {
```

Comment:
Makes a little more sense to me to check that the output of the command that you depend on (e.g., `$certThumbprint`) looks sane (I guess just non-empty in this case) instead of the exit code of the last command, but maybe you need both?

Reply:
Agreed, modified this to be similar to the bash version where the value becomes `NOCERT` on failure of the keyvault command.